tests/block_tools.py

```diff
@@ -850,12 +855,15 @@ def get_consecutive_blocks(
                             else:
                                 pool_target = PoolTarget(self.pool_ph, uint32(0))
                         if transaction_data is not None:
-                            if previous_generator is not None and type(previous_generator) is CompressorArg:
-                                block_generator = best_solution_generator_from_template(
+                            if type(previous_generator) is CompressorArg:
+                                block_generator: Optional[BlockGenerator] = best_solution_generator_from_template(
                                     previous_generator, transaction_data
                                 )
                             else:
                                 block_generator = simple_solution_generator(transaction_data)
+                                if type(previous_generator) is list:
+                                    block_generator = BlockGenerator(block_generator.program, [], previous_generator)
+
```

Comment:
this is the second place this check had to be added to. This is to fix the existing feature where `previous_generator` is allowed to be a list of heights.